### PR TITLE
ENH: Update percentile dask chunking

### PR DESCRIPTION
**Pull request by bzah**
### Pull Request Checklist:
- [x] This PR addresses an already opened issue (for bug fixes / features)
    - This PR is related to #932 
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Documentation has been added / updated (for bug fixes / features)
- [x] HISTORY.rst has been updated (with summary of main changes)
  - [x] Link to issue (:issue:`number`) and pull request (:pull:`number`) has been added.
- [ ] `bumpversion patch` has been called on this branch
- [ ] The relevant author information has been added to `.zenodo.json`

### What kind of change does this PR introduce?
This PR make sure `percentile_doy` rechunking preserve the initial chunk size.

Currently, ``percentile_doy`` multiply the size of chunks by at least `window` width (default 5).
This size increase can even be larger because the time dimension initial chunking is lost in percentile_doy rechunking.
This PR fix this by chunking ``dayofyear`` dimension in `window * time_chunk_count` chunks.

Using dask dashboard I see the memory peak use being reduced by about 30% for a mono-threaded launch with theses changes,
However, I expected a bigger reduction of memory use.
The performances seems to also improve, even though dask exhibits a new warning: ``PerformanceWarning: Slicing with an out-of-order index is generating 36 times more chunks``
I'm not sure what this warning is about, it comes from [dask slicing].(https://github.com/dask/dask/blob/main/dask/array/slicing.py#L630)

I suspect the performance improvement is mainly due to fewer I/O tasks for spilling and reading memory on disk.

<details>
<summary>Test Script</summary>

```
client = Client(memory_limit='16GB', n_workers=1, threads_per_worker=1)
big_tas_file = "../icclim/netcdf_files/tas_day_INM-CM5-0_ssp585_r1i1p1f1_gr1_20650101-21001231.nc"
da = xr.open_dataset(big_tas_file).tas
dask.config.set({"array.chunk-size": f"{200} MB"})
chunking = {d: "auto" for d in da.dims}
# chunking["time"] = -1
da = da.chunk(chunking)

da_per = da.sel(time=slice("2065-01-01", "2094-12-31"))
tas_per = percentile_doy(da_per, window=5, per=90)
a = xc.atmos.tg90p(tas=da, t90=tas_per, freq="MS", bootstrap=False)
a.compute()
```
</details>

### Does this PR introduce a breaking change?
No.

### Other information:
I'm putting this in draft for now as I would like to do a few more tests.
Any feedback would be also appreciated.

### CODE REVIEW

**Comment by bzah** [comment]
Fichtre. 
It doesn't work when input has leap years. `adjust_doy_calendar` calls a apply_ufunc `dayofyear` as a core dim...

**Comment by bzah** [comment]
Leaving draft!
Performances and memory looks good. 
Christian is doing a few more test with real data but I think it can be reviewed as is.

**Review by huard — APPROVED** [review]
Looks good. @aulemahal any thoughts ?

File: xclim/core/calendar.py [hunk 1]
```diff
@@ -619,12 +621,14 @@ def _interpolate_doy_calendar(source: xr.DataArray, doy_max: int) -> xr.DataArra
     doy_max_source = int(source.dayofyear.max())
 
     # Interpolate to fill na values
-    tmp = source.interpolate_na(dim="dayofyear")
+    filled_na = source.chunk(dict(dayofyear=-1)).interpolate_na(dim="dayofyear")
```
[the remark this answers is not in the document]
Comment (bzah) [reply xclim/core/calendar.py:624]: Good catch thank you.

**Comment by Zeitsperre** [comment]
Feel free to merge when you're ready! Thanks again!

**Comment by bzah** [comment]
Thanks guys for the review !

**Pull request merged**